selfapi.js

```diff
@@ -6,6 +6,11 @@ var https = require('https');
 var nodepath = require('path');
 var url = require('url');
 
+var options = {
```

Comment:
Nit: Please export these options so that API implementors can modify them, i.e. `exports.options = {` (using `exports.options` instead of just `options` in the rest of the code) or `var options = exports.options = {` (using `options` everywhere as you do now). I don't have a strong preference for either.